trino/trino.go

```diff
@@ -1344,6 +1344,13 @@ func (qr *driverRows) fetch() error {
 				err = io.EOF
 			} else if err == context.Canceled {
 				qr.Close()
+			} else {
+				var queryErr *ErrQueryFailed
+				if errors.As(err, &queryErr) {
+					if errors.Is(queryErr.Reason, context.DeadlineExceeded) {
+						qr.Close()
+					}
+				}
 			}
```

Comment:
This can be simplified because `ErrQueryFailed` implements the unwrap interface. From https://pkg.go.dev/errors:
> [Is](https://pkg.go.dev/errors#Is) examines the tree of its first argument looking for an error that matches the second. It reports whether it finds a match. It should be used in preference to simple equality checks

```suggestion
			} else if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
				qr.Close()
			}
```